test/drivers/start_tests.cpp

```diff
@@ -0,0 +1,273 @@
+	oargs.engine_name = UNKNOWN_ENGINE;
+	oargs.mode = SCAP_MODE_LIVE;
+	unsigned long buffer_bytes_dim = DEFAULT_DRIVER_BUFFER_BYTES_DIM;
+	std::string kmod_path;
+
+	/* Remove kmod if injected, we remove it always even if we use another engine
+	 * in this way we are sure the unique driver in the system is the one we will use.
+	 */
+	if(remove_kmod())
+	{
+		return EXIT_FAILURE;
+	}
+
+	/* Get current cwd */
+	char cwd[FILENAME_MAX];
+	if(!getcwd(cwd, FILENAME_MAX))
+	{
+		std::cerr << "Unable to get current dir" << std::endl;
+		return EXIT_FAILURE;
+	}
+
+	/* Parse CLI options */
+	int op = 0;
+	int long_index = 0;
+	while((op = getopt_long(argc, argv,
+				"b::mk::d:",
+				long_options, &long_index)) != -1)
+	{
+		switch(op)
+		{
+		case 'b':
+			oargs.engine_name = BPF_ENGINE;
+			bpf_params.buffer_bytes_dim = buffer_bytes_dim;
+			if(optarg == NULL)
+			{
+				bpf_params.bpf_probe = strncat(cwd, BPF_PROBE_DEFAULT_PATH, FILENAME_MAX - strlen(cwd));
+			}
+			else
+			{
+				bpf_params.bpf_probe = optarg;
+			}
+			oargs.engine_params = &bpf_params;
+
+			/* The BPF `calibrate_socker` method needs to call the socket filler
+			 * before starting the real capture. So we attach all syscalls and
+			 * `sys_enter` and `sys_exit` tracepoints.
+			 */
+			oargs.tp_of_interest.tp[SYS_ENTER] = 1;
+			oargs.tp_of_interest.tp[SYS_EXIT] = 1;
+			for(int i = 0; i < PPM_SC_MAX; i++)
+			{
+				oargs.ppm_sc_of_interest.ppm_sc[i] = 1;
+			}
+
+			std::cout << "* Configure BPF probe tests! Probe path: " << bpf_params.bpf_probe << std::endl;
+			break;
+
+		case 'm':
+			oargs.engine_name = MODERN_BPF_ENGINE;
+			modern_bpf_params.buffer_bytes_dim = buffer_bytes_dim;
+			oargs.engine_params = &modern_bpf_params;
+			std::cout << "* Configure modern BPF probe tests!" << std::endl;
+			break;
+
+		case 'k':
+			oargs.engine_name = KMOD_ENGINE;
+			kmod_params.buffer_bytes_dim = buffer_bytes_dim;
+			if(optarg == NULL)
+			{
+				kmod_path = strncat(cwd, KMOD_DEFAULT_PATH, FILENAME_MAX - strlen(cwd));
+			}
+			else
+			{
+				kmod_path = optarg;
+			}
+			oargs.engine_params = &kmod_params;
+			if(insert_kmod(kmod_path))
+			{
+				return EXIT_FAILURE;
+			}
+			std::cout << "* Configure kernel module tests! Kernel module path: " << kmod_path << std::endl;
+			;
+			break;
+
+		case 'd':
+			buffer_bytes_dim = strtoul(optarg, NULL, 10);
+			break;
+
+		default:
+			break;
+		}
+	}
+	std::cout << "* Using buffer dim: " << buffer_bytes_dim << std::endl;
+
+	if(strcmp(oargs.engine_name, UNKNOWN_ENGINE) == 0)
+	{
+		std::cerr << "Unsupported engine! Choose between: m, b, k" << std::endl;
+		return EXIT_FAILURE;
+	}
+
+	char error_buffer[FILENAME_MAX] = {0};
+	event_test::scap_handle = scap_open(&oargs, error_buffer, &ret);
+	if(!event_test::scap_handle)
+	{
+		std::cerr << "Unable to open the engine: " << error_buffer << std::endl;
+		return EXIT_FAILURE;
+	}
+	return EXIT_SUCCESS;
+}
+
+void print_setup_phase_message()
```

Comment:
Stupid nit: i think we can safely refactor the `print_X_message` to a single function that prints a `const char *` .